kubos/update.py

```diff
@@ -0,0 +1,107 @@
+# Kubos SDK
+# Copyright (C) 2016 Kubos Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import git
+import json
+import logging
+import sys
+import time
+import os
+
+from kubos.utils.git_common import *
+from kubos.version import check_provided_version
+from kubos import version as kubos_version
+from packaging import version
+from yotta import link, link_target
+from yotta.lib import detect
+from yotta.options import parser
+
+def addOptions(parser):
+    pass
+
+
+def execCommand(args, following_args):
+    if os.geteuid() != 0:
+        #After changing the version the new target/modules are re-linked to /usr/local/lib.. which requires root to succeed
+        print 'Woops, this command needs to be run as root. Please re-run `sudo kubos update`'
```

Comment:
Yes, It will fail if it's not run as root (unless we go and change permissions for `/usr/local/*`. If we made root the default user we could remove this but that would be playing with fire a little bit.